opm/grid/cpgrid/Intersection.hpp

```diff
@@ -300,6 +301,12 @@ namespace Dune
                 return *this;
             }
 
+            IntersectionIterator operator++(int)
+            {
+                IntersectionIterator tmp(*this);
```

Comment:
This implementation is wrong. The correct is to increment this object (call increment() here) after you make the copy, and then return the copy in `tmp`. As is, you leave this object unchanged, but return an incremented object.

Reply:
Good catch, thank you!